.github/agents/component-author.agent.md

```diff
@@ -0,0 +1,533 @@
+      this.input.type = 'text';
+      this.append(this.input);
+    }
+
+    this.setFormControlElement(this.input);
+  }
+}
+```
+
+### Component with Scoped Elements
+
+```typescript
+import { type ScopedElementsMap, ScopedElementsMixin } from '@open-wc/scoped-elements/lit-element.js';
+import { Icon } from '@sl-design-system/icon';
+import { Button } from '@sl-design-system/button';
+
+export class ComponentName extends ScopedElementsMixin(LitElement) {
+  /** @internal */
+  static get scopedElements(): ScopedElementsMap {
+    return {
+      'sl-icon': Icon,
+      'sl-button': Button
+    };
+  }
+}
+```
+
+### Component with Localization
+
+```typescript
+import { localized, msg } from '@lit/localize';
+
+@localized()
+export class ComponentName extends LitElement {
+  override render(): TemplateResult {
+    return html`
+      <button aria-label=${msg('Close dialog')}>
+        ${msg('Close')}
+      </button>
+    `;
+  }
+}
+```
+
+## Coding Conventions
+
+### Critical Rules
+1. **Use `.js` extensions** in imports (code runs in browser)
+   ```typescript
+   import { Component } from './component.js';
+   import styles from './component.scss.js';
+   ```
+
+2. **TypeScript for types only** - Use for annotations and interfaces
+
+3. **Styles in SCSS** - Place in `<component>.scss` files
+   - Build automatically transforms to `<component>.scss.ts`
+   - Import as `styles` from the `.scss.js` file
+
+4. **Property Order**
+   - Public properties before private properties
+   - Public methods before private methods
+   - Private methods/properties use `#` prefix
+
+5. **Property Decorators**
+   - Use `@property({ type: Boolean, reflect: true })` for boolean props
+   - Use `@property({ reflect: true })` for string props that should reflect to attributes
+   - Use `@property()` for non-reflecting properties
+   - Always include JSDoc with `@default` values
+
+6. **Lifecycle Methods**
+   - `connectedCallback()` - Setup, add event listeners
+   - `disconnectedCallback()` - Cleanup
+   - `firstUpdated()` - One-time initialization after first render
+   - `updated()` - React to property changes
+   - `willUpdate()` - Before rendering (compute derived state)
+
+### Type Definitions
+
+Always export type unions for component variants:
+```typescript
+export type ComponentSize = 'sm' | 'md' | 'lg';
+export type ComponentVariant = 'primary' | 'secondary' | 'danger';
+export type ComponentFill = 'solid' | 'outline' | 'ghost';
+```
+
+### Global Type Declarations
+
+Always declare global types:
+```typescript
+declare global {
+  interface HTMLElementTagNameMap {
+    'sl-component': ComponentName;
+  }
+
+  // For custom events
+  interface GlobalEventHandlersEventMap {
+    'sl-custom': SlCustomEvent;
+  }
+}
+```
+
+## Accessibility Guidelines
+
+1. **Semantic HTML** - Use appropriate semantic elements
+2. **ARIA Attributes** - Add proper roles, labels, and states
+3. **Keyboard Navigation** - Implement arrow keys, Enter, Space, Escape
```

Comment:
Maybe sth like: `RovingTabindexController` should be used for the arrow keys navigation